examples/esp-todomvc-react/js/todoItem.jsx

```diff
@@ -0,0 +1,111 @@
+/*global React */
+var app = app || {};
+
+(function () {
+    "use strict";
+
+    var ESCAPE_KEY = 27,
+        ENTER_KEY = 13;
+
+    app.TodoItem = React.createClass({
+
+        getInitialState: function () {
+            return {editText: ""};
+        },
+
+        handleDestroy: function () {
+            this.props.router.publishEvent("todoList", "todoDestroyed", { id: this.props.id });
```

Comment:
I briefly considered creating some helper func on components e.g. `this.publish('event', {...})` but decided against it because I thought it would mask the API and make the example less clear...